app/controllers/users/sessions_controller.rb

```diff
@@ -10,6 +10,16 @@ class Users::SessionsController < Devise::SessionsController
   before_action :redirect_to_dashboard, only: %i[sign_in_with_token redirect_from_magic_link]
   before_action :ensure_login_token_valid, only: %i[sign_in_with_token redirect_from_magic_link]
 
+  def new
+    super do
+      if flash.present?
+        flash.clear
+        resource.valid?
+        resource.errors.delete(:full_name)
+      end
+    end
```

Comment:
What does this syntax do? Call the body first and then the `new` method in the parent?

Reply:
The method definition in super is 
```
  def new
    self.resource = resource_class.new(sign_in_params)
    clean_up_passwords(resource)
    yield resource if block_given?
    respond_with(resource, serialize_options(resource))
  end
```

So it just allows us to inject a bit of our own code in the middle there